### PR TITLE
[docs] Fix wrong attributes in lambda_permission doc

### PR DESCRIPTION
**Pull request by spesnova**
## Reasoning for docs update
Examples in the doc uses wrong attributes:

- `${aws_lambda_function.test_lambda.arn}` as lambda's function name
- `${aws_iam_role.default.arn}` as IAM role name

The doc should be use `${aws_lambda_function.test_lambda.function_name}` as lambda's function name and  `"${aws_iam_role.default.name}` as IAM role name.

## Relevant Terraform version
All versions which support `aws_lambda_permission` resource.

### CODE REVIEW

File: website/source/docs/providers/aws/r/lambda_permission.html.markdown [hunk 1]
```diff
@@ -27,14 +27,14 @@ resource "aws_lambda_permission" "allow_cloudwatch" {
...
-  function_name    = "${aws_lambda_function.test_lambda.arn}"
+  function_name    = "${aws_lambda_function.test_lambda.function_name}"
   function_version = "$LATEST"
 }
 
 resource "aws_lambda_function" "test_lambda" {
   filename      = "lambdatest.zip"
   function_name = "lambda_function_name"
-  role          = "${aws_iam_role.iam_for_lambda.arn}"
+  role          = "${aws_iam_role.iam_for_lambda.name}"
```
Comment (radeksimko) [inline website/source/docs/providers/aws/r/lambda_permission.html.markdown]: I agree the attribute name may be confusing, however the API in fact rejects names and only accepts ARNs, see what happens when you try to apply the changed example:

```
$ terraform apply
aws_iam_role.iam_for_lambda: Creating...
  arn:                "" => "<computed>"
  assume_role_policy: "" => "{\n  \"Version\": \"2012-10-17\",\n  \"Statement\": [\n    {\n      \"Action\": \"sts:AssumeRole\",\n      \"Principal\": {\n        \"Service\": \"lambda.amazonaws.com\"\n      },\n      \"Effect\": \"Allow\",\n      \"Sid\": \"\"\n    }\n  ]\n}\n"
  create_date:        "" => "<computed>"
  name:               "" => "iam_for_lambda_rsimko"
  path:               "" => "/"
  unique_id:          "" => "<computed>"
aws_iam_role.iam_for_lambda: Creation complete (ID: iam_for_l...a_rsimko)
aws_lambda_function.test_lambda: Creating...
  arn:              "" => "<computed>"
  filename:         "" => "../lambdatest.zip"
  function_name:    "" => "lambda_function_name_rsimko"
  handler:          "" => "exports.handler"
  last_modified:    "" => "<computed>"
  memory_size:      "" => "128"
  publish:          "" => "false"
  qualified_arn:    "" => "<computed>"
  role:             "" => "iam_for_lambda_rsimko"
  runtime:          "" => "nodejs4.3"
  source_code_hash: "" => "<computed>"
  timeout:          "" => "3"
  version:          "" => "<computed>"
Error applying plan:

1 error(s) occurred:

* aws_lambda_function.test_lambda: 1 error(s) occurred:

* aws_lambda_function.test_lambda: Error creating Lambda function: ValidationException: 1 validation error detected: Value 'iam_for_lambda_rsimko' at 'role' failed to satisfy constraint: Member must satisfy regular expression pattern: arn:aws:iam::\d{12}:role/?[a-zA-Z_0-9+=,.@\-_/]+
	status code: 400, request id: f37875c2-1516-11e7-8be2-a1a8bd2483c5

```

File: website/source/docs/providers/aws/r/lambda_permission.html.markdown [hunk 2]
```diff
@@ -83,7 +83,7 @@ resource "aws_sns_topic_subscription" "lambda" {
 resource "aws_lambda_function" "func" {
   filename      = "lambdatest.zip"
   function_name = "lambda_called_from_sns"
-  role          = "${aws_iam_role.default.arn}"
+  role          = "${aws_iam_role.default.name}"
```
Comment (radeksimko) [inline website/source/docs/providers/aws/r/lambda_permission.html.markdown]: See above.

File: website/source/docs/providers/aws/r/lambda_permission.html.markdown [hunk 3]
```diff
@@ -27,14 +27,14 @@ resource "aws_lambda_permission" "allow_cloudwatch" {
 resource "aws_lambda_alias" "test_alias" {
   name             = "testalias"
   description      = "a sample description"
-  function_name    = "${aws_lambda_function.test_lambda.arn}"
+  function_name    = "${aws_lambda_function.test_lambda.function_name}"
```
Comment (radeksimko) [inline website/source/docs/providers/aws/r/lambda_permission.html.markdown]: The alias accepts either name or the ARN (i.e. both), which is [also documented](https://www.terraform.io/docs/providers/aws/r/lambda_alias.html#function_name), so this is not a real issue, but I agree we can change it and make it look cleaner. 👍

File: website/source/docs/providers/aws/r/lambda_permission.html.markdown [hunk 4]
```diff
@@ -65,7 +65,7 @@ EOF
 resource "aws_lambda_permission" "with_sns" {
   statement_id  = "AllowExecutionFromSNS"
   action        = "lambda:InvokeFunction"
-  function_name = "${aws_lambda_function.my-func.arn}"
+  function_name = "${aws_lambda_function.my-func.function_name}"
```
Comment (radeksimko) [inline website/source/docs/providers/aws/r/lambda_permission.html.markdown]: While this is _not_ documented, the API accepts either ARN or name here - happy to accept this change though, as it makes it look cleaner. 👍

File: website/source/docs/providers/aws/r/lambda_permission.html.markdown (hunk 3 above)
Comment (spesnova) [inline website/source/docs/providers/aws/r/lambda_permission.html.markdown]: Oh, I didn't know that the ARN can be used for it.

> but I agree we can change it and make it look cleaner. 👍

Thanks!

File: website/source/docs/providers/aws/r/lambda_permission.html.markdown (hunk 1 above)
Comment (spesnova) [inline website/source/docs/providers/aws/r/lambda_permission.html.markdown]: Yes, you are right.

I found it on the aws-sdk-go. 

```golang
     // The Amazon Resource Name (ARN) of the IAM role that Lambda assumes when it
     // executes your function to access any other Amazon Web Services (AWS) resources.
     Role *string `type:"string"`

```

https://github.com/aws/aws-sdk-go/blob/6d13e167d1db68c8ef8c1b27066bcd03457d22e3/service/lambda/api.go#L3940-L3942

I'll return here to `arn`.

File: website/source/docs/providers/aws/r/lambda_permission.html.markdown (hunk 2 above)
Comment (spesnova) [inline website/source/docs/providers/aws/r/lambda_permission.html.markdown]: 👌

File: website/source/docs/providers/aws/r/lambda_permission.html.markdown (hunk 4 above)
Comment (spesnova) [inline website/source/docs/providers/aws/r/lambda_permission.html.markdown]: 👍

**Comment by spesnova** [comment]
> With that said it's currently already broken because of missing runtime attribute in aws_lambda_function. Would you mind adding it?

Sure 🙆‍♂️ 
I noticed that `runtime` attribute is missing in [the doc](https://www.terraform.io/docs/providers/aws/r/lambda_function.html) for `aws_lambda_function` resource too. Should I add it to the doc?

**Comment by spesnova** [comment]
@radeksimko 

updated.

If I should update the doc for `aws_lambda_function`, please let me know. I'll update it too.

**Comment by radeksimko** [comment]
Thanks for the updates, pulling in...

> I noticed that runtime attribute is missing in the doc for aws_lambda_function resource too. Should I add it to the doc?

Sure, feel free to send a PR 👍  😸

**Pull request merged**